plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -486,7 +492,19 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
                 properties.buildOrThrow(),
                 Optional.ofNullable(tableHandle.getMetadataEntry().getDescription()),
                 constraints.stream()
-                        .map(SparkExpressionParser::toTrinoExpression)
+                        .map(constraint -> {
+                            try {
+                                return toTrinoExpression(constraint);
+                            }
+                            catch (TrinoException e) {
+                                if (e.getErrorCode().equals(DELTA_LAKE_UNSUPPORTED_EXPRESSION.toErrorCode()) && suppressExpressionParserFailure) {
```

Comment:
Is comparing such generic error type as USER_ERROR safe here? 
What if someone else uses this error type for for other error?

Is this logic actually needed? Seems that `toTrinoExpression` can fail only because of failed conversion.

Reply:
> Is comparing such generic error type as USER_ERROR safe here?
What if someone else uses this error type for for other error?

Even if someone uses `DELTA_LAKE_UNSUPPORTED_EXPRESSION` in different places, that doesn't break this logic. If someone change the exception type inside of  `toTrinoExpression`, we can catch by CI failure. 

> Is this logic actually needed? Seems that toTrinoExpression can fail only because of failed conversion.

What's your suggestion? If we remove this block, SELECT query fails even though check constraints shouldn't affect it.

Reply:
> DELTA_LAKE_UNSUPPORTED_EXPRESSION in different places, that doesn't break this logic.

I was rather referring to someone creating other exception type and using USER_ERROR there. With newest changes any DELTA_LAKE_* error will be suppressed.

My suggestion was to not check exception error type at all and treat all `TrinoException` as parsing error, because `toTrinoExpression` only parses constraint. Even when somone add another exception there and will use `ErrorType.EXTERNAL` errorCode, exception will be supressed.